app/controllers/admin/members_controller.rb

```diff
@@ -1,12 +1,44 @@
+require "csv"
+
 module Admin
   class MembersController < ::MembersController
-    http_basic_authenticate_with MembershipRegister.basic_auth_options
-
-    respond_to :csv, only: :index
+    http_basic_authenticate_with(
+      name: ENV.fetch("AUTH_NAME") { "secretary" },
+      password: ENV.fetch("AUTH_PASSWORD") {
+        password = SecureRandom.hex
+        Rails.logger.info("You have not set a password for the admin area, your temporary password is: #{password}")
+        password
+      }
+    )
 
     def index
       @members = Member.all.order(updated_at: :desc)
-      respond_with @members
+
+      respond_to do |format|
+        format.html
+        format.csv do
+          csv_content = CSV.generate(headers: true) do |csv|
```

Comment:
What about just `Member.pluck(:id, :full_name, ...).to_csv`?

Reply:
> What about just `Member.pluck(:id, :full_name, ...).to_csv`?

That is a great idea :clap::clap::clap: